graph/flow/interval_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package flow
+
+import (
+	"testing"
+
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+func TestInterval(t *testing.T) {
+	// Graph from C. Cifuentes, "Reverse Compilation Techniques", 1994 (figure 6-23)
+	g := simple.NewDirectedGraph()
+	edges := []struct{ from, to int }{
+		{1, 2},
+		{1, 5},
+		{2, 3},
+		{2, 4},
+		{3, 5},
+		{4, 5},
+		{5, 6},
+		{6, 7},
+		{6, 12},
+		{7, 8},
+		{7, 9},
+		{8, 9},
+		{8, 10},
+		{9, 10},
+		{10, 11},
+		{12, 13},
+		{14, 13},
+		{13, 14},
+		{14, 15},
+		{15, 6},
+	}
+
+	for _, e := range edges {
+		g.SetEdge(g.NewEdge(simple.Node(e.from), simple.Node(e.to)))
+	}
+
+	// test number of intervals
+	ig := Intervals(g, 1)
+	if len(ig.Intervals) != 3 {
+		t.Fatalf("Expected 3 intervals, got %d", len(ig.Intervals))
+	}
+
+	// test number of nodes
```

Comment:
Consider extending the test cases to also verify that the correct nodes are added to each interval.